.github/workflows/build.yml

```diff
@@ -12,18 +12,54 @@ on:
     types: [published]
 
 jobs:
-  build:
-    if: |
+  smoke-tests:
+    if: | 
       github.event_name == 'push'
       || github.event_name == 'release'
       || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository)
-
+    
     outputs:
       GIT_TAG: ${{ steps.variables.outputs.GIT_TAG }}
       GIT_BRANCH: ${{ steps.variables.outputs.GIT_BRANCH }}
       OUTPUT_DIR: ${{ steps.variables.outputs.OUTPUT_DIR }}
+    
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout code
+        uses: actions/checkout@v3
+      -
+        name: "Calculate required variables"
+        id: variables
+        run: |
+          GIT_TAG=${{ github.event.release.tag_name }}
+          # If GIT_TAG is set then GIT BRANCH should be "master", else set it from GITHUB_REF
+          GIT_BRANCH=$([ -n "${GIT_TAG}" ] && echo "master" || echo "${GITHUB_REF#refs/*/}")
+          echo ::set-output name=GIT_BRANCH::${GIT_BRANCH}
+          echo ::set-output name=GIT_TAG::${GIT_TAG}
+          echo ::set-output name=OUTPUT_DIR::${GIT_TAG:-${GIT_BRANCH}}
+      -
+        name: "Check git branch name depth"
+        env:
+          GIT_BRANCH: ${{ steps.variables.outputs.GIT_BRANCH }}
+        run: |
+          IFS='/';
+          read -r -a branch <<<"${GIT_BRANCH}";
+          if [[ "${#branch[@]}" -gt 2 ]]; then echo "Error: Your branch name contains more than one subdir, which will cause issues with the build process." && FAIL=1; fi;
+          unset IFS;
+          # If FAIL is 1 then we fail.
+          [[ $FAIL == 1 ]] && exit 1 || echo "Branch name depth check passed."
+        shell: bash
+
+  build:
+    if: |
```

Comment:
`needs: smoke-tests` should negate the need for this `if` block

Reply:
True. Leftover from copy&pasting.